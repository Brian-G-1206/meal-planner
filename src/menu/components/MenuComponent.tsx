import { Form, Formik, FormikProps } from "formik"
import React, { FunctionComponent } from "react"
import { FormSelectOption } from "../../controls/formSelect/FormSelect"

export interface MenuProps {
	dinnerOptions: FormSelectOption[],
}

export const MenuComponent: FunctionComponent<MenuProps & FormikProps<MenuFormValues>> = ( props ) => 
<div>
	<h1>Menu</h1>
	<Formik
		initialValues = {{ sundayDinner: "" }}
		onSubmit={(values, actions) => {}}
	>
	</Formik>
</div>