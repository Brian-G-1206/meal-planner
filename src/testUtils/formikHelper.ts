import { FormikProps } from "formik";

export const formikMenuFormValuesTestProps: FormikProps<MenuFormValues> = {
	values: {
		sundayDinner: '',
	},
	errors: {},
	touched: {},
	isSubmitting: false,
	isValidating: false,
	submitCount: 0,
	setStatus: () => {},
	setSubmitting: () => {},
	setTouched: () => {},
	setErrors: () => {},
	setValues: () => {},
	setFieldError: () => {},
	setFieldValue: () => {},
	setFieldTouched: () => {},
	setFormikState: () => {},
	validateField: () => {},
	resetForm: () => {},
	submitForm: async () => {},
	validateForm: async (values: any) => ({
		sundayDinner: '',
	}),
	handleSubmit: () => {},
	handleBlur: () => {},
	handleReset: () => {},
	handleChange: () => {},
	dirty: false, 
	initialTouched: {},
	isValid: true,
	initialValues: {
		sundayDinner: '',
	},
	initialErrors: {},
	//I have no idea if any of this is right, I just want to run some tests
	registerField: () => {},
	unregisterField: () => {},
	getFieldProps: <Value = any>( props: any) => ({name: '', onChange: () => {}, onBlur: () => {}, value: props}),
	getFieldMeta: <Value>(name: any) => ({name, onChange: () => {}, onBlur: () => {}, value: name, touched: false, initialTouched: false }),
	getFieldHelpers: <Value>(name: any) => ({name, onChange: () => {}, onBlur: () => {}, value: name, touched: false, initialTouched: false, setValue: () => {}, setTouched: () => {}, setError: () => {} }),

}