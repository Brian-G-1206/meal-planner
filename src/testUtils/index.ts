export const getEmptyRootState = () : RootState => ({
	menu: getEmptyMenuState()
})

export const getEmptyMenuState = () : MenuState => ({
	dinnerOptions : [

	]
})