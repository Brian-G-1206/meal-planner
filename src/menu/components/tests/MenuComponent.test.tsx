import { shallow, ShallowWrapper } from 'enzyme'
import renderer from 'react-test-renderer'
import { formikMenuFormValuesTestProps } from '../../../testUtils/formikHelper'
import { MenuComponent } from '../MenuComponent'

describe('src/menu/components', () => {
	describe('MenuComponent', () => {
		describe('rendering', () => {
			it('should render the form', () => {
				const Subject = renderer
				.create(<MenuComponent  dinnerOptions={[{ label: "", value: "" }]} {...formikMenuFormValuesTestProps } />)
				.toJSON()
				expect(Subject).toMatchSnapshot()
			})
		})

		describe('onSubmit', () => {
			it('should do nothing ... yet', () => {
				const Subject: ShallowWrapper = shallow(<MenuComponent  dinnerOptions={[{ label: "", value: "" }]} {...formikMenuFormValuesTestProps } />)
				const form = Subject.find('Formik')
				form.simulate('submit')
				
			})
		})
	})
})