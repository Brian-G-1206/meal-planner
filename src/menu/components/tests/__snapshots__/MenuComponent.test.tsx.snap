// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/menu/components MenuComponent rendering should render the form 1`] = `
<div>
  <h1>
    Menu
  </h1>
</div>
`;
